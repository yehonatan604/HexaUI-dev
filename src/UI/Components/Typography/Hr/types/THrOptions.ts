import { TComponentOptions } from "../../../../../Data/Types/TComponent";

export type THrOptions = TComponentOptions & {
    color?: string;
    width?: string;
    height?: string;
}
