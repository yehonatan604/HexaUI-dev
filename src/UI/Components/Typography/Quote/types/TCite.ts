export type TCite = {
    url: string;
    text: string;
    placement: "start" | "end" | "center";
    variant: string;
};
