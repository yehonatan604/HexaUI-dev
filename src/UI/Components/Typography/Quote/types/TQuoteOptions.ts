import { TComponentOptions } from "../../../../../Data/Types/TComponentOptions";

export type TQuoteOptions = TComponentOptions & {
    gap?: number;
}
