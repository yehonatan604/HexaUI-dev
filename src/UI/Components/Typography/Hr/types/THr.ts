import { TComponent } from "../../../../../Data/Types/TComponent";
import { THrOptions } from "./THrOptions";

export type THr = TComponent<"hr"> & {
    options?: THrOptions;
}
