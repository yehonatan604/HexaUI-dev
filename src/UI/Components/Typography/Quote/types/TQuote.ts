import { TComponent } from "../../../../../Data/Types/TComponent";
import { TQuoteOptions } from "./TQuoteOptions";
import { TCite } from "./TCite";

export type TQuote = TComponent<"blockquote"> & {
    options: TQuoteOptions;
    cite: TCite;
}
