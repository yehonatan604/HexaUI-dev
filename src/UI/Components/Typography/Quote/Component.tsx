import useTheme from "../../../../Core/Hooks/useTheme";
import { FlexDir } from "../../../../Data/Constants/FlexDirection";
import { FlexTypes } from "../../../../Data/Constants/FlexTypes";
import { TQuote } from "../../../../Data/Types/ComponentTypes/Typography/Quote/TQuote";
import Flex from "../../Layout/Flex/Component";

const Quote = (props: TQuote) => {
  const { children, cite, options, className, ...componenetProps } = props;

  // Hooks
  const { colors } = useTheme();

  // Colors
  const quoteColor = options?.textVariant || colors.standrad;
  const citeColor = cite.variant || colors.standrad;
  const flexGap = options?.gap || 4;

  // JSX
  return (
    <>
      <blockquote className={`${className} m-auto`} {...componenetProps}>
        <Flex
          options={{
            direction: FlexDir.Col,
            align:
              cite.placement === "start"
                ? FlexTypes.Start
                : cite.placement === "end"
                ? FlexTypes.End
                : FlexTypes.Center,
          }}
          className={`gap-${flexGap} w-1/2 m-auto`}
        >
          <p className={`text-${quoteColor} text-pretty font-serif italic`}>
            " {children} "
          </p>
          <a href={cite.url} className={`text-sm text-${citeColor} hover:opacity-70`}>
            {cite.text}
          </a>
        </Flex>
      </blockquote>
    </>
  );
};

export default Quote;
