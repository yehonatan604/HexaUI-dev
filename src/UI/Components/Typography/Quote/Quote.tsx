import { FlexDir } from "../../../../Data/Constants/FlexDirection";
import { FlexTypes } from "../../../../Data/Constants/FlexTypes";
import Flex from "../../Layout/Flex/Flex";
import { TQuote } from "./types/TQuote";

const Quote = (props: TQuote) => {
  const { children, cite, options, className, ...componenetProps } = props;

  // Colors
  const quoteColor = options?.textVariant || "standard";
  const citeColor = cite.variant || "standard";
  const flexGap = options?.gap || 4;

  // JSX
  return (
    <>
      <blockquote className={`${className} m-auto`} {...componenetProps}>
        <Flex
          role="presentation"
          options={{
            direction: FlexDir.Col,
            align:
              cite.placement === "start"
                ? FlexTypes.Start
                : cite.placement === "end"
                ? FlexTypes.End
                : FlexTypes.Center,
          }}
          className={`gap-${flexGap} w-1/2 m-auto`}
        >
          <p className={`text-${quoteColor} text-pretty font-serif italic`}>
            " {children} "
          </p>
          <a href={cite.url} className={`text-sm text-${citeColor} hover:opacity-70`}>
            {cite.text}
          </a>
        </Flex>
      </blockquote>
    </>
  );
};

export default Quote;
