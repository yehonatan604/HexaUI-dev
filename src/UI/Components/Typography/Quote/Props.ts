import { TCite } from "../../../../Data/Types/ComponentTypes/Typography/Quote/TCite";
import { TComponent } from "../../../../Data/Types/ComponentTypes/TComponent";

export type QuoteProps = TComponent<"blockquote"> & {
    cite: TCite;
    gap?: string;
};