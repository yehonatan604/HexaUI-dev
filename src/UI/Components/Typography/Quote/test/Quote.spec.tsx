import { render, screen } from "@testing-library/react";
import Quote from "../Quote"; // Adjust the import path
import { describe, it, expect } from "vitest";
import { TCite } from "../types/TCite";

describe("Quote Component", () => {
  it("should render the quote text and citation", () => {
    render(
      <Quote
        cite={
          {
            url: "https://example.com",
            text: "Source",
            placement: "end",
            variant: "blue-500",
          } as TCite & string
        }
        options={{}}
      >
        This is a test quote.
      </Quote>
    );

    // Check if the quote text is rendered
    expect(screen.getByText(/This is a test quote./)).toBeInTheDocument();

    // Check if the citation is rendered and has the correct link
    const citeElement = screen.getByText("Source");
    expect(citeElement).toBeInTheDocument();
    expect(citeElement).toHaveAttribute("href", "https://example.com");
  });

  it("should apply custom text and cite variants", () => {
    render(
      <Quote
        cite={
          {
            url: "https://example.com",
            text: "Source",
            placement: "end",
            variant: "red-500",
          } as TCite & string
        }
        options={{ textVariant: "green-500" }}
      >
        This is a custom colored quote.
      </Quote>
    );

    const quoteText = screen.getByText(/This is a custom colored quote./);
    const citeElement = screen.getByText("Source");

    // Check if the custom text color for the quote and citation are applied
    expect(quoteText).toHaveClass("text-green-500");
    expect(citeElement).toHaveClass("text-red-500");
  });

  it("should align the citation based on the placement option", () => {
    render(
      <Quote
        cite={
          {
            url: "https://example.com",
            text: "Source",
            placement: "start",
            variant: "blue-500",
          } as TCite & string
        }
        options={{}}
      >
        This is a test quote.
      </Quote>
    );

    const flexContainer = screen.getByRole("presentation");

    // Check if the citation alignment is correct
    expect(flexContainer).toHaveClass("items-start");
  });

  it("should render and match the snapshot", () => {
    const { asFragment } = render(
      <Quote
        cite={
          {
            url: "https://example.com",
            text: "Source",
            placement: "end",
            variant: "blue-500",
          } as TCite & string
        }
        options={{ textVariant: "green-500", gap: 2 }}
      >
        This is a snapshot test quote.
      </Quote>
    );

    // Snapshot the Quote structure
    expect(asFragment()).toMatchSnapshot();
  });
});
