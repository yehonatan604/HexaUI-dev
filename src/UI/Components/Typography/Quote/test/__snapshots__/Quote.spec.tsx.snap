// Vitest Snapshot v1

exports[`Quote Component > should render and match the snapshot 1`] = `
<DocumentFragment>
  <blockquote
    class="undefined m-auto"
  >
    <div
      class="flex flex-col bg- text- justify-start items-end flex-nowrap gap-2 w-1/2 m-auto"
      role="presentation"
    >
      <p
        class="text-green-500 text-pretty font-serif italic"
      >
        " This is a snapshot test quote. "
      </p>
      <a
        class="text-sm text-blue-500 hover:opacity-70"
        href="https://example.com"
      >
        Source
      </a>
    </div>
  </blockquote>
</DocumentFragment>
`;
